scapy/libs/extcap.py

```diff
@@ -0,0 +1,241 @@
+# SPDX-License-Identifier: GPL-2.0-only
+# This file is part of Scapy
+# See https://scapy.net/ for more information
+# Copyright (C) Gabriel Potter
+
+"""
+Wireshark extcap API utils
+https://www.wireshark.org/docs/wsdg_html_chunked/ChCaptureExtcap.html
+"""
+
+import collections
+import functools
+import pathlib
+import re
+import subprocess
+
+from scapy.config import conf
+from scapy.consts import WINDOWS
+from scapy.data import MTU
+from scapy.error import warning
+from scapy.interfaces import (
+    network_name,
+    resolve_iface,
+    InterfaceProvider,
+    NetworkInterface,
+)
+from scapy.packet import Packet
+from scapy.supersocket import SuperSocket
+from scapy.utils import PcapReader, _create_fifo, _open_fifo
+
+# Typing
+from typing import (
+    cast,
+    Any,
+    Dict,
+    List,
+    NoReturn,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+)
+
+
+def _extcap_call(prog: str,
+                 args: List[str],
+                 format: Dict[str, List[str]],
+                 ) -> Dict[str, List[Tuple[str, ...]]]:
+    """
+    Function used to call a program using the extcap format,
+    then parse the results
+    """
+    p = subprocess.Popen(
+        [prog] + args,
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        text=True
+    )
+    data, err = p.communicate()
+    if p.returncode != 0:
+        raise OSError("%s returned with error code %s: %s" % (prog, p.returncode, err))
+    res = collections.defaultdict(list)
```

Comment:
Could we use a namedtuple here? That could help improve code readability.

Reply:
I tried for a bit and I find it more confusing.
I didn't find a clean way of iterating over the named values of a namedtuple so it makes it pretty cumbersome here.